plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -1178,3 +1288,235 @@ SELECT
+	rs.max_log_bytes_used as [max_log_bytes_used],
+	rs.avg_log_bytes_used as avg_log_bytes_used
+FROM sys.query_store_query q WITH (NOLOCK)
+	JOIN sys.query_store_plan p WITH (NOLOCK) ON q.query_id = p.query_id
+	JOIN sys.query_store_runtime_stats rs WITH (NOLOCK) ON p.plan_id = rs.plan_id
+	JOIN sys.query_store_runtime_stats_interval rsi WITH (NOLOCK) ON rs.runtime_stats_interval_id = rsi.runtime_stats_interval_id
+WHERE rsi.start_time >= @currIntervalStartDate AND rsi.end_time <= @currIntervalEndDate;
+'
+
+EXEC #ExecForeachDb @SqlText
+
+SELECT * FROM #QueryStoreMetrics;
+
+SELECT CONVERT(NVARCHAR(MAX),(SELECT * FROM #DbTimestamps FOR XML PATH('ts'), ROOT('Timestamps'))) AS CachedData;
+`
+
+const sqlAzureMIQueryStoreWaitStatistics = sqlAzureMIPartQueryPeriod + `
+IF OBJECT_ID ('tempdb.dbo.#QueryStoreWaitStat') IS NOT NULL DROP TABLE #QueryStoreWaitStat;
+CREATE TABLE #QueryStoreWaitStat(
+	measurement varchar(39),
+	measurement_id varchar(36),
+	sql_instance nvarchar(256),
+	[database_name] nvarchar(128),
+	interval_start_time datetime,
+	interval_end_time datetime,
+	interval_start_time_d bigint,
+	interval_end_time_d bigint,
+	query_id bigint,
+	query_hash nvarchar(max),
+	plan_id bigint,
+	last_compile_batch_sql_handle nvarchar(max),
+	query_plan_hash nvarchar(max),
+	wait_category_s nvarchar(60),
+	exec_type_d tinyint,
+	total_query_wait_time_ms_d bigint,
+	max_query_wait_time_ms_d bigint,
+	query_param_type_d tinyint
+);
+
+SET @SqlText = @SqlText + N'
+INSERT INTO #QueryStoreWaitStat
+SELECT
+	''sqlserver_azuremi_querystore_wait_stats'' AS [measurement],
+	CONVERT(varchar(36), NEWID()) AS [measurement_id],
+	REPLACE(@@SERVERNAME, ''\'', '':'') AS [sql_instance],
+	DB_NAME() AS [database_name],
+	CONVERT(datetime, rsi.start_time, 1) AS interval_start_time,
+	CONVERT(datetime, rsi.end_time, 1) AS interval_end_time,
+	CONVERT(BIGINT, DATEDIFF(day, 0, CONVERT(datetime, rsi.start_time, 1))) * @magicNumberDays + DATEDIFF(second, 0, DATEADD(day,-DATEDIFF(day, 0, CONVERT(datetime, rsi.start_time, 1)), CONVERT(datetime, rsi.start_time, 1))) * @magicNumberSeconds AS interval_start_time_d,
+	CONVERT(BIGINT, DATEDIFF(day, 0, CONVERT(datetime, rsi.end_time, 1))) * @magicNumberDays + DATEDIFF(second, 0, DATEADD(day,-DATEDIFF(day, 0, CONVERT(datetime, rsi.end_time, 1)), CONVERT(datetime, rsi.end_time, 1))) * @magicNumberSeconds AS interval_end_time_d,
+	q.query_id,
+	CONVERT(nvarchar(max), q.query_hash, 1) as query_hash,
+	p.plan_id,
+	CONVERT(nvarchar(max), q.last_compile_batch_sql_handle, 1) as last_compile_batch_sql_handle,
+	CONVERT(nvarchar(max), p.query_plan_hash, 1) as query_plan_hash,
+	ws.wait_category_desc AS wait_category_s,
+	ws.execution_type AS exec_type_d,
+	ws.total_query_wait_time_ms AS total_query_wait_time_ms_d,
+	ws.max_query_wait_time_ms AS max_query_wait_time_ms_d,
+	q.query_parameterization_type AS query_param_type_d
+FROM sys.query_store_query q WITH (NOLOCK)
+	JOIN sys.query_store_plan p WITH (NOLOCK) ON q.query_id = p.query_id
```

Comment:
Do we need sys.query-store_plan? only column is plan_id which is in the waitstats DMV.